money-core/src/test/scala/com/comcast/money/core/FormattersSpec.scala

```diff
@@ -16,40 +16,128 @@
 
 package com.comcast.money.core
 
+import java.util.UUID
+
 import com.comcast.money.api.SpanId
 import org.scalatest.{ Matchers, WordSpec }
+import Formatters._
+import org.scalacheck.Arbitrary
+import org.scalacheck.Test.Failed
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+
+class FormattersSpec extends WordSpec with Matchers with GeneratorDrivenPropertyChecks with TraceGenerators {
 
-class FormattersSpec extends WordSpec with Matchers {
+  implicit val arbitraryUUID: Arbitrary[UUID] = Arbitrary(genUUID)
 
-  private val expectedTraceId = "a"
-  private val expectedParentSpanId = "1"
-  private val expectedSpanId = "2"
   "Http Formatting" should {
-    "convert from a money  http header" in {
-      val spanId = new SpanId()
-      val test = Formatters.toHttpHeader(spanId)
-      Formatters.fromHttpHeader(test).get shouldBe spanId
+    "read a money  http header" in {
+      forAll { (traceIdValue: UUID, parentSpanIdValue: Long, spanIdValue: Long) =>
+        val expectedSpanId = new SpanId(traceIdValue.toString, parentSpanIdValue, spanIdValue)
+        val spanId = fromMoneyHeader(header => header match {
+          case MoneyTraceHeader => MoneyHeaderFormat.format(expectedSpanId.traceId, expectedSpanId.parentId, expectedSpanId.selfId)
+        }, _ => {})
```

Comment:
I would prefer a structure more like this without the default declared and not using the match block. A partial function will serve the same purpose and it's less verbose:
````    
"fail to read a badly formatted money  http header" in {
      forAll { (traceIdValue: String, parentSpanIdValue: String, spanIdValue: String) =>
        val spanId =
          fromMoneyHeader(
            getHeader = {
            case MoneyTraceHeader => MoneyHeaderFormat.format(traceIdValue, parentSpanIdValue, spanIdValue)
          }
          )

        spanId shouldBe None
      }
    }
```